x-pack/plugins/infra/public/hooks/use_fetch_triggered_alert_history.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AsApiContract } from '@kbn/actions-plugin/common';
+import { HttpSetup } from '@kbn/core/public';
+import {
+  ALERT_DURATION,
+  ALERT_RULE_UUID,
+  ALERT_START,
+  ALERT_STATUS,
+  ALERT_TIME_RANGE,
+} from '@kbn/rule-data-utils';
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+
+interface UseFetchTriggeredAlertsHistoryProps {
+  features: string;
+  ruleId: string;
+}
+interface FetchTriggeredAlertsHistory {
+  totalTriggeredAlerts: number;
+  histogramTriggeredAlerts: Array<{
+    key_as_string: string;
+    key: number;
+    doc_count: number;
+  }>;
+  error?: string;
+  avgTimeToRecoverUS: number;
+}
+
+interface TriggeredAlertsHistory {
+  isLoadingTriggeredAlertHistory: boolean;
+  errorTriggeredAlertHistory?: string;
+  triggeredAlertsData?: FetchTriggeredAlertsHistory;
+}
+export function useFetchTriggeredAlertsHistory({
```

Comment:
**suggestion:** this hook reminds me of a similar one we already have in the `infra` plugin that we use to fetch the active alerts count, here are some improvements you could apply:
- Rename to `useAlertsHistory`, what you want to return is not a fetcher handler but a dataset, so this naming seems more concise
- The `${BASE_RAC_ALERTS_API_PATH}/find` API has been improved to accept the `features_id` option in case the index is not known. This is a great optimization because it allows triggering a single request to perform the search instead of sequentially asking for the index before being able to make the request.
- The state management for this hook could be simplified using the existing `useAsyncFn` hook from `react-use`, which abstracts some logic out of the box related to the meta state.
- The accepted `features` params could be an array in case we want to filter the subset of alerts by more domains, and you can directly forward it to the `_find` API as described above.
- Would be nice to test this hook to verify its correct behaviour is always maintained, [here an example](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/hooks/use_alerts_count.test.ts) of how the useAlertsCount hook is tested.

These suggestions come from the [`useAlertsCount`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/hooks/use_alerts_count.ts) implementation, let me know if you have any doubt about it.

Reply:
Thanks for the changes! Would you also switch the manual state management for the `useAsyncFn` hook? I believe it'll help to make the code more readable and easier to maintain.
Ok for the test, you could create an issue to not lose track of that pending task.

Reply:
Regarding using the `useAsyncFn ` https://github.com/elastic/kibana/issues/154170